### PR TITLE
fix sporadic 503s

### PR DESCRIPTION
**Pull request by lloyd**
A fix for train-2013.04.26 with changelog update.

@jrgm - ohai!

### CODE REVIEW

**Comment by jaredhirsch** [comment]
@lloyd super super excited about relaxing consistency requirements via this fix :beers:

File: lib/wsapi.js [hunk 1]
```diff
@@ -112,7 +122,18 @@ function authenticateSession(options, cb) {
...
     }
     cb(null);
-  });
+  }
+
+  // if the client provided last reset timestamp, use that.  otherwise,
+  // hit the database.
+  if (typeof lastReset === 'number') {
+    process.nextTick(function() {
+      console.log("here I am");
```
Comment (jaredhirsch) [inline lib/wsapi.js]: looks like a leftover console.log

File: lib/wsapi/auth_with_assertion.js [hunk 2]
```diff
@@ -106,6 +106,7 @@ exports.process = function(req, res) {
           r = r.body;
           if (!r.success) throw "non-success response from dbwriter";
           if (!r.userid) throw "malformed response from dbwriter";
+          if (!r.lastPasswordReset) throw "malformed response from dbwriter";
```
Comment (jaredhirsch) [inline lib/wsapi/auth_with_assertion.js]: can we make these two error messages distinct for debugging purposes?

**Comment by jaredhirsch** [comment]
R+ from me, modulo few nits, but @jrgm seems to have more context and should clicky the green button

**Comment by jaredhirsch** [comment]
@lloyd forgot to namecheck you in the above comments

**Comment by jrgm** [comment]
Yeah, I'm good with this for train-2013.04.26, with @6a68's two nits fixed. And you'll need to update the browserid.spec from '2' to '3'.

**Pull request merged**